#import "gRPCDelegate.h"
#import "viewercontroller.pb.h"

@interface gRPCViewerControllerDelegate : NSObject

delegateDeclaration(ViewerControllerCloseViewer, ViewerController, Response)
delegateDeclaration(ViewerControllerPixList, ViewerControllerPixListRequest, ViewerControllerPixListResponse)
delegateDeclaration(ViewerControllerNeedsDisplayUpdate, ViewerController, Response)
delegateDeclaration(ViewerControllerROIList, ViewerControllerROIListRequest, ViewerControllerROIListResponse)
delegateDeclaration(ViewerControllerNewROI, ViewerControllerNewROIRequest, ViewerControllerNewROIResponse)
delegateDeclaration(ViewerControllerCurDCM, ViewerController, ViewerControllerCurDCMResponse)
delegateDeclaration(ViewerControllerROIsWithName, ViewerControllerROIsWithNameRequest, ViewerControllerROIsWithNameResponse)
delegateDeclaration(ViewerControllerSelectedROIs, ViewerController, ViewerControllerSelectedROIsResponse)
delegateDeclaration(ViewerControllerIsDataVolumic, ViewerControllerIsDataVolumicRequest, ViewerControllerIsDataVolumicResponse)
delegateDeclaration(ViewerControllerCopyViewerWindow, ViewerController, ViewerControllerCopyViewerWindowResponse)
delegateDeclaration(ViewerControllerResampleViewerController, ViewerControllerResampleViewerControllerRequest, ViewerControllerResampleViewerControllerResponse)
delegateDeclaration(ViewerControllerBlendingController, ViewerController, ViewerControllerBlendingControllerResponse)
delegateDeclaration(ViewerControllerVRControllers, ViewerController, ViewerControllerVRControllersResponse)
delegateDeclaration(ViewerControllerTitle, ViewerController, ViewerControllerTitleResponse)
delegateDeclaration(ViewerControllerModality, ViewerController, ViewerControllerModalityResponse)
delegateDeclaration(ViewerControllerMovieIdx, ViewerController, ViewerControllerMovieIdxResponse)
delegateDeclaration(ViewerControllerSetMovieIdx, ViewerControllerSetMovieIdxRequest, Response)
delegateDeclaration(ViewerControllerMaxMovieIdx, ViewerController, ViewerControllerMaxMovieIdxResponse)
delegateDeclaration(ViewerControllerIdx, ViewerController, ViewerControllerIdxResponse)
delegateDeclaration(ViewerControllerSetIdx, ViewerControllerSetIdxRequest, Response)
delegateDeclaration(ViewerControllerDisplayedIdx, ViewerController, ViewerControllerDisplayedIdxResponse)
delegateDeclaration(ViewerControllerSetDisplayedIdx, ViewerControllerSetDisplayedIdxRequest, Response)
delegateDeclaration(ViewerControllerWLWW, ViewerController, ViewerControllerWLWWResponse)
delegateDeclaration(ViewerControllerSetWLWW, ViewerControllerSetWLWWRequest, Response)
delegateDeclaration(ViewerControllerOpenVRViewerForMode, ViewerControllerOpenVRViewerForModeRequest, ViewerControllerOpenVRViewerForModeResponse)
delegateDeclaration(ViewerControllerFuseWithViewer, ViewerControllerFuseWithViewerRequest, Response)

@end
